### PR TITLE
V4.40.0 pre1 linux

### PR DESCRIPTION
**Pull request by SplinterGU**
add linux support

### CODE REVIEW

**Comment by monkeyman192** [comment]
You can't remove the windows targets and replace with Linux ones, you need to add Linux as well as windows

**Comment by SplinterGU** [comment]
I don't remove windows target... look again... it generate windows and linux... but, only net6.0.... it works... look...
image
[out.txt](https://github.com/monkeyman192/MBINCompiler/files/12469687/out.txt)

File: MBINCompiler/MBINCompiler.csproj [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>net6.0-windows;net7.0-windows</TargetFrameworks>
```
Comment (monkeyman192) [inline MBINCompiler/MBINCompiler.csproj:4]: you removed the `net7.0-windows` target here. You need to keep all the `net7.0` targets

File: MBINCompiler/MBINCompiler.csproj [hunk 2]
```diff
@@ -29,23 +29,23 @@
...
+    <RuntimeIdentifier>linux-x64;win-x64</RuntimeIdentifier>
     <PlatformTarget>x64</PlatformTarget>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
     <OutputPath>..\Build\Release\</OutputPath>
     <DefineConstants>TRACE</DefineConstants>
-    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
+    <RuntimeIdentifier>linux-x64;win-x64</RuntimeIdentifier>
```
Comment (monkeyman192) [inline MBINCompiler/MBINCompiler.csproj]: If you are adding another identifier, the tag should be `RuntimeIdentifiers`, cf. MBINCompilerTests/MBINCompilerTests.csproj:L7

**Comment by monkeyman192** [comment]
In terms of the target frameworks we specify, I am not sure if it's best to always specify the operating system in the target frameworks, or whether if we just have `net6.0;net7.0` as the target frameworks and then we have both linux and windows as the runtimes we target whether that will work just fine.

**Comment by SplinterGU** [comment]
I already added net7.0

you can do a merge (with fixes) if you want

File: MBINCompiler/MBINCompiler.csproj (hunk 1 above)
Comment (SplinterGU) [reply MBINCompiler/MBINCompiler.csproj:4]: already added

File: MBINCompiler/MBINCompiler.csproj (hunk 2 above)
Comment (SplinterGU) [reply MBINCompiler/MBINCompiler.csproj]: oh, I understand... wait...

Comment (SplinterGU) [reply MBINCompiler/MBINCompiler.csproj]: done! thanks!

**Comment by SplinterGU** [comment]
> 

yes, only net6.0 and net7.0 is enough... I'll change it... wait...

File: build-net6.bat
Comment (monkeyman192) [inline build-net6.bat:1]: Realistically, I'd say that the .bat files would only be run on windows, and the .sh file would only be run on linux, so it probably makes sense to only build for the runtime specific to that platform (ie. .bat file just just `-r win-x64`, and the .sh file has just `-r linux-x64`)

Comment (SplinterGU) [reply build-net6.bat:1]: done! fixed!

well, I don't know if it must be 1 or several script... now works...

Comment (SplinterGU) [reply build-net6.bat:1]: now I can build from linux, windows and linux target...

Comment (monkeyman192) [reply build-net6.bat:1]: The files still build both linux and windows.
I think it should be that the .bat is a single line to build windows, and .sh is a single line to build linux.

**Comment by monkeyman192** [comment]
One final thing before I allow the CI to run, you need to modify the `pipeline.yml` file to provide the correct framework.
You can also add a separate step to build linux if you want, or leave that to me.

File: build-net6.bat
Comment (SplinterGU) [reply build-net6.bat:1]: is your project, change it as you wish... In my projects I have an build all, for build all from 1 host plattform... but you do as you wish... it is only a merge request... you edit it if you wish...

**Comment by SplinterGU** [comment]
> One final thing before I allow the CI to run, you need to modify the `pipeline.yml` file to provide the correct framework. You can also add a separate step to build linux if you want, or leave that to me.

I'm ignorant about this, I don't know how to do it... that's why I wanted to send you the changes through Discord... but well, undoubtedly this branching thing saves you much more work than passing it through Discord...

**Comment by monkeyman192** [comment]
> I'm ignorant about this, I don't know how to do it... that's why I wanted to send you the changes through Discord... but well, undoubtedly this branching thing saves you much more work than passing it through Discord...

It's just a text file. Have a look at lines 26 - 30. They need to have the frameworks changes so that they are just `net6.0` and `net7.0` instead of `...-windows`

**Comment by SplinterGU** [comment]
> > I'm ignorant about this, I don't know how to do it... that's why I wanted to send you the changes through Discord... but well, undoubtedly this branching thing saves you much more work than passing it through Discord...
> 
> It's just a text file. Have a look at lines 26 - 30. They need to have the frameworks changes so that they are just `net6.0` and `net7.0` instead of `...-windows`

ok, let me check... question, is possible add a zip and a tgz for download all binaries files in a click?

**Comment by monkeyman192** [comment]
> > > I'm ignorant about this, I don't know how to do it... that's why I wanted to send you the changes through Discord... but well, undoubtedly this branching thing saves you much more work than passing it through Discord...
> > 
> > 
> > It's just a text file. Have a look at lines 26 - 30. They need to have the frameworks changes so that they are just `net6.0` and `net7.0` instead of `...-windows`
> 
> ok, let me check... question, is possible add a zip and a tgz for download all binaries files in a click?

yes, but I can do that for you as I think I'll need to rejig the pipeline a bit to add the linux build, and if you don't know how to do that it will be easier for me to just do that once we have the build profiles set up at least for local building.

**Comment by SplinterGU** [comment]
> > > > I'm ignorant about this, I don't know how to do it... that's why I wanted to send you the changes through Discord... but well, undoubtedly this branching thing saves you much more work than passing it through Discord...
> > > 
> > > 
> > > It's just a text file. Have a look at lines 26 - 30. They need to have the frameworks changes so that they are just `net6.0` and `net7.0` instead of `...-windows`
> > 
> > 
> > ok, let me check... question, is possible add a zip and a tgz for download all binaries files in a click?
> 
> yes, but I can do that for you as I think I'll need to rejig the pipeline a bit to add the linux build, and if you don't know how to do that it will be easier for me to just do that once we have the build profiles set up at least for local building.

I didn't need to install anything cross-platform; simply installing dotnet-sdk-6.0 and dotnet-sdk-7.0 was enough for me.

**Pull request closed without merging**